pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go

```diff
@@ -500,16 +492,25 @@ func labelSelectors(useNamespaceSelector bool) (namespaceSelector, objectSelecto
 
 // mutatePodExecOptions handles mutating PodExecOptions and encoding and decoding admission
 // requests and responses for the public mutate functions
-func mutatePodExecOptions(rawPodExecOptions []byte, name string, ns string, userInfo *authenticationv1.UserInfo, m mutatePodExecFunc, dc dynamic.Interface, apiClient kubernetes.Interface) ([]byte, error) {
+func mutatePodExecOptions(rawPodExecOptions []byte, name string, ns string, mutationType string, userInfo *authenticationv1.UserInfo, m mutatePodExecFunc, dc dynamic.Interface, apiClient kubernetes.Interface) ([]byte, error) {
 	var exec corev1.PodExecOptions
 	if err := json.Unmarshal(rawPodExecOptions, &exec); err != nil {
 		return nil, fmt.Errorf("failed to decode raw object: %v", err)
 	}
 
-	if err := m(&exec, name, ns, userInfo, dc, apiClient); err != nil {
+	if _, err := m(&exec, name, ns, userInfo, dc, apiClient); err != nil {
 		return nil, err
 	}
 
+	injected, err := m(&exec, name, ns, userInfo, dc, apiClient)
+
+	if err != nil {
+		metrics.MutationErrors.Inc(mutationType, err.Error())
+		return nil, err
+	}
+
+	metrics.MutationAttempts.Inc(mutationType, strconv.FormatBool(injected))
```

Comment:
should we remove the increment here since it is already done in the `common.go` file

Reply:
I don't think so, because increment is done in the `common.go` file in the generic `Mutate` [function](https://github.com/DataDog/datadog-agent/blob/20bd520300827f097aca9f4eb056e585568a0888/pkg/clusteragent/admission/mutate/common/common.go#L29).

All webhooks use this generic Mutate function to build the webhook mutation entry point and wrap the injection logic (examples: [agent_sidecar](https://github.com/DataDog/datadog-agent/blob/20bd520300827f097aca9f4eb056e585568a0888/pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go#L108-L111), [auto_instrumentation](https://github.com/DataDog/datadog-agent/blob/20bd520300827f097aca9f4eb056e585568a0888/pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go#L272-L275), ), except for the cws command injection webhook, which uses this `mutatePodExecOptions` function and doesn't pass through the generic `Mutate`.

This is why incrementing is necessary in this function.